idea/src/org/jetbrains/kotlin/idea/intentions/RemoveBracesIntention.kt

```diff
@@ -61,8 +63,16 @@ class RemoveBracesIntention : SelfTargetingIntention<KtElement>(KtElement::class
 
         val newElement = block.replace(statement.copy())
 
+        val factory = KtPsiFactory(block)
+
         if (construct is KtDoWhileExpression) {
-            newElement.parent!!.addAfter(KtPsiFactory(block).createNewLine(), newElement)
+            newElement.parent!!.addAfter(factory.createNewLine(), newElement)
+        }
+
+        if (construct is KtIfExpression &&
+            container.node.elementType == KtNodeTypes.ELSE &&
+            construct.getStrictParentOfType<KtDotQualifiedExpression>() != null) {
```

Comment:
Why do you consider just this case? What if we have (e.g.) KtSafeQualifiedExpression instead?

Reply:
I'd say any `KtExpression` at next level should work here.